execution_engine/src/storage/trie_store/operations/mod.rs

```diff
@@ -394,6 +479,238 @@ where
+                            Trie::Extension {
+                                affix: extension_affix,
+                                pointer,
+                            } => {
+                                let mut new_affix = vec![sibling_idx];
+                                new_affix.extend(Vec::<u8>::from(extension_affix));
+                                let new_extension: Trie<K, V> = Trie::Extension {
+                                    affix: new_affix.into(),
+                                    pointer,
+                                };
+                                let trie_key = Blake2bHash::new(&new_extension.to_bytes()?);
+                                new_elements.push((trie_key, new_extension))
+                            }
+                        }
+                    }
+                }
+            }
+            // The parent is a pointer block, and we are propagating a node or extension upwards. It
+            // is impossible to propagate a leaf upwards.  Reseat the thing we are propagating into
+            // the parent.
+            (Some((trie_key, _)), Trie::Node { mut pointer_block }) => {
+                let trie_node: Trie<K, V> = {
+                    pointer_block[idx as usize] = Some(Pointer::NodePointer(*trie_key));
+                    Trie::Node { pointer_block }
+                };
+                let trie_key = Blake2bHash::new(&trie_node.to_bytes()?);
+                new_elements.push((trie_key, trie_node))
+            }
+            // The parent is an extension, and we are outputting an extension.  Prepend the
+            // parent affix to affix of the output extension, mutating the output in place.
+            // This is the only mutate-in-place.
+            (
+                Some((
+                    trie_key,
+                    Trie::Extension {
+                        affix: child_affix,
+                        pointer,
+                    },
+                )),
+                Trie::Extension { affix, .. },
+            ) => {
+                let mut new_affix: Vec<u8> = Vec::<u8>::from(affix);
+                new_affix.extend(Vec::<u8>::from(child_affix.to_owned()));
+                *child_affix = new_affix.into();
+                *trie_key = {
+                    let new_extension: Trie<K, V> = Trie::Extension {
+                        affix: child_affix.to_owned(),
+                        pointer: pointer.to_owned(),
+                    };
+                    Blake2bHash::new(&new_extension.to_bytes()?)
+                }
+            }
+            // The parent is an extension and the new element is a pointer block.
+            // The next element we add will be an extension to the pointer block we are going to
+            // add.
+            (Some((trie_key, Trie::Node { .. })), Trie::Extension { affix, mut pointer }) => {
+                pointer = Pointer::NodePointer(*trie_key);
```

Comment:
It would be less confusing to just not bind that pointer, since we're not actually mutating the parent.
```suggestion
            (Some((trie_key, Trie::Node { .. })), Trie::Extension { affix, .. }) => {
                let pointer = Pointer::NodePointer(*trie_key);
```

Reply:
Fixed in https://github.com/CasperLabs/casper-node/pull/1155/commits/a2723a86125857bad79405a86dc66e52ca680154